lib/view_model/exchange/exchange_view_model.dart

```diff
@@ -77,10 +96,20 @@ abstract class ExchangeViewModelBase with Store {
   final Box<Trade> trades;
   final ExchangeTemplateStore _exchangeTemplateStore;
   final TradesStore tradesStore;
+  final SharedPreferences sharedPreferences;
 
   @observable
   ExchangeProvider provider;
 
+  /// Maps in dart are not sorted by default
```

Comment:
I added documentation for most of the changes here, yet feel free to comment on anything that is unclear or needs more details